src/react-components/avatar-selector.js

```diff
@@ -0,0 +1,119 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { injectIntl, FormattedMessage } from 'react-intl';
+
+class AvatarSelector extends Component {
+  static propTypes = {
+    avatars: PropTypes.array,
+    avatar: PropTypes.string,
+    onChange: PropTypes.func,
+  }
+
+  getAvatarIndex(direction=0) {
+    const currAvatarIndex = this.props.avatars.findIndex(avatar => avatar.id === this.props.avatar);
+    const numAvatars = this.props.avatars.length;
+    return ((currAvatarIndex + direction) % numAvatars + numAvatars) % numAvatars;
+  }
+
+  nextAvatar = () => {
+    const newAvatarIndex = this.getAvatarIndex(1);
```

Comment:
I'd probably extract two oneliner methods `nextAvatarIndex` and `previousAvatarIndex`